singularity-eos/eos/eos_gruneisen.cpp

```diff
@@ -12,12 +12,155 @@
 // publicly and display publicly, and to permit others to do so.
 //------------------------------------------------------------------------------
 
+#include <sstream>
+
+#include <root-finding-1d/root_finding.hpp>
 #include <singularity-eos/eos/eos.hpp>
 
 namespace singularity {
 
 PORTABLE_INLINE_FUNCTION Real square(const Real x) { return x * x; }
 PORTABLE_INLINE_FUNCTION Real cube(const Real x) { return x * x * x; }
+PORTABLE_INLINE_FUNCTION Real find_min_bounded_val(const Real root1, const Real root2,
+                                                   const Real min, const Real max) {
+  // Try to find the minimum bounded value. If none exists, return a value less than the
+  // minimum
+  const Real minroot = std::min(root1, root2);
+  const Real maxroot = std::max(root1, root2);
+  if (minroot > min && minroot < max) {
+    return minroot;
+  } else if (maxroot > min && maxroot < max) {
+    return maxroot;
+  } else {
+    return min - 1.;
```

Comment:
how should I interpret this else case? Is this a fail case? If so perhaps we should return a specific flag number, like signal_nan?

Reply:
I might have coded this to be a little too general here, because my intent was to just fail at an arbitrary number outside of the bounds on the minimum side. Since I'm only operating on `eta` in compression, any negative value would suffice. I'll change this.

Reply:
> 👍 Maybe add a comment explaining the logic as well.

There is a comment at the top of the function explaining the behavior. Were you wanting something more verbose?

Reply:
nah just a comment at this else case like "any negative number will do" or something like that.